core/src/main/java/hudson/util/RingBufferLogHandler.java

```diff
@@ -94,16 +94,29 @@ public synchronized void clear() {
      * <p>
      * New records are always placed early in the list.
      */
-    public synchronized List<LogRecord> getView() {
-        List<LogRecord> result = new ArrayList<>(size);
-        for (int i = 0; i < size; i++) {
-            LogRecord lr = records[(start + i) % records.length].get();
-            if (lr != null) {
-                result.add(lr);
+    public List<LogRecord> getView() {
+        // Since Jenkins.logRecords is a field used as an API, we are forced to implement a dynamic list.
+        return new AbstractList<LogRecord>() {
+            @Override
+            public LogRecord get(int index) {
+                // flip the order
+                synchronized (RingBufferLogHandler.this) {
+                    LogRecord r = records[(start + (size - (index + 1))) % records.length].get();
+                    // We cannot just omit collected entries due to the List interface.
+                    return r != null ? r : new LogRecord(Level.OFF, "<discarded>");
```

Comment:
@timja  after this change is our log full with this messages.

![image](https://user-images.githubusercontent.com/89339813/165734835-c79081ec-c029-4887-a093-ac05e5d90054.png)

 How can I deactivate it?
![image](https://user-images.githubusercontent.com/89339813/165735290-a2754794-74cb-4217-8dca-33a14a5df509.png)

Reply:
This can happen if memory is full and garbage collection discards old messages. There is not a configuration option. Anyway I would not suggest using the GUI logger screen for real tracking; you can install the `support-core` plugin to get proper log files written to disk (and rotated).

Reply:
Thx, we will try it!